components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -1297,6 +1308,179 @@ public void changeApplicationOwner(String applicationId, ApplicationOwner applic
         updateServiceProvider(applicationId, appToUpdate);
     }
 
+    /**
+     * Authorize an API resource to the application.
+     *
+     * @param applicationId              Application ID.
+     * @param authorizedAPICreationModel API Authorization creation model.
+     */
+    public void addAuthorizedAPI(String applicationId, AuthorizedAPICreationModel authorizedAPICreationModel) {
+
+        try {
+            String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            String authorizedAPIId = authorizedAPICreationModel.getId();
+            AuthorizedAPI authorizedAPI = getAuthorizedAPIManagementService().getAuthorizedAPI(applicationId,
+                    authorizedAPIId, tenantDomain);
+            if (authorizedAPI != null) {
+                throw handleAuthorizedAPIConflictError(applicationId, authorizedAPIId);
+            }
+
+            // Validate authorized API creation model.
+            APIResource apiResource = ApplicationManagementServiceHolder.getApiResourceManager()
+                    .getAPIResourceById(authorizedAPIId, tenantDomain);
+            if (apiResource == null) {
+                throw buildClientError(ErrorMessage.API_RESOURCE_NOT_FOUND, authorizedAPIId, tenantDomain);
+            }
+            validateAPIResourceScopes(apiResource, authorizedAPICreationModel.getScopes());
+
+            // Validate policy identifier.
+            String policyIdentifier = validatePolicy(authorizedAPICreationModel.getPolicyIdentifier());
+
+            // If API resource has requiresAuthorization set to true, policy identifier should be RBAC.
+            if (apiResource.isRequiresAuthorization() &&
```

Comment:
```suggestion
            if (apiResource.isAuthorizationRequired() &&
```
can we rename the property as isAuthorizationRequired

Reply:
This needs an identity framework level change to the APIResource object.